### PR TITLE
LAMP KB RTDs

### PR DESCRIPTION
**Pull request by sfsyunusSLAC**
- Added RTDs
- Added interlock to stop bender motor motion if T > 100C

### CODE REVIEW

File: lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU [hunk 1]
```diff
@@ -440,6 +564,12 @@ nEncRefXM3K4  := ULINT_TO_UDINT(GVL_M3K4_Constants.nM3K4X_ENC_REF);
 nEncRefYM3K4  := ULINT_TO_UDINT(GVL_M3K4_Constants.nM3K4Y_ENC_REF);
 nEncRefrXM3K4 := ULINT_TO_UDINT(GVL_M3K4_Constants.nM3K4rX_ENC_REF);
 
+// LAMP KB Bender RTD interlocks
+M10.bHardwareEnable := nM2K4US_RTD_1 < 1000;
+M11.bHardwareEnable := nM2K4DS_RTD_1 < 1000;
+M15.bHardwareEnable := nM3K4US_RTD_1 < 1000;
+M16.bHardwareEnable := nM3K4DS_RTD_1 < 1000;
```
Comment (ZLLentz) [inline lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU:571]: Is there an error bit available? I suspect this interlock fails if the RTD is disconnected and showing 0. Something like:
`M16.bHardwareEnable := nM3K4DS_RTD_1 < 1000 AND NOT bM3K4DS_RTD_1_Err`

Comment (sfsyunusSLAC) [reply lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU:571]: A disconnected RTD reads value = 8500
Would that work?
`M16.bHardwareEnable := nM3K4DS_RTD_1 < 1000 AND NOT 8500`

Comment (slacAWallace) [reply lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU:571]: `M16.bHardwareEnable R= nM3K4DS_RTD_1 > 1000 OR bM3K4DS_RTD_1_Err`

R= means reset the left side of the operator if the right side is true. Useful when you want to set something false, but not necessarily set it true.

Comment (sfsyunusSLAC) [reply lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU:571]: Just noticed my comparison operator is going the wrong way!

Comment (ZLLentz) [reply lcls-plc-tmo-optics/tmo_optics/POUs/MAIN.TcPOU]: There's no reason to make a special case if the error value is also over 1000

I'm just trying to cover all the edge cases

Like what if the IO is not linked? Is the hardware enable-able in that case?

**Pull request merged**